qiskit_experiments/base_analysis.py

```diff
@@ -63,13 +77,18 @@ def run(
                 f"Invalid experiment data type, expected {self.__experiment_data__.__name__}"
                 f" but received {type(experiment_data).__name__}"
             )
+        # Get analysis options
+        analysis_options = self._default_options()
+        analysis_options.update_options(**options)
+        analysis_options = analysis_options.__dict__
+
         # Run analysis
         # pylint: disable=broad-except
         try:
-            analysis_results, figures = self._run_analysis(experiment_data, **options)
+            analysis_results, figures = self._run_analysis(experiment_data, **analysis_options)
             analysis_results["success"] = True
-        except Exception:
-            analysis_results = AnalysisResult(success=False)
+        except Exception as ex:
+            analysis_results = AnalysisResult(success=False, error_message=ex)
```

Comment:
I wonder if this is too broad? I would suggest to catch only an `AnalysisError` and the `_run_analysis` method of the subclasses will need to raise these errors when appropriate. This could be for a follow-up PR.

Reply:
I've got a separate PR open for this: #63